### ISSUE TITLE
Auto scroll down (or) collapse other tree views on expanding the node inside a tree view with multiple tree views in display instead of manually scrolling down

### ISSUE THREAD

**Issue by sreeharshakakani** [eclipse-theia/theia#11572]
Add auto scroll functionality on the current tree view (or) collapse the other expanded tree view in the explorer to make more UI space to view on expanding the node with multiple tree views in display.

VS Code Version: Version: 1.70.1
OS Version: MacOS Big Sur Version: 11.5.2
Steps to Reproduce:

Cloned the tree-view-sample vscode extension repo : [tree-view-sample](https://github.com/microsoft/vscode-extension-samples/tree/main/tree-view-sample)
1) Run these commands : npm install
2) npm run watch
3) F5 to start debugging
4) Once all the multiple tree views are displayed in the package explorer section, expand each individual tree view with expanding individual nodes inside the tree view

On expanding the node inside a tree view should either collapse other tree view which are currently not in use (or) auto scroll down giving more space on the UX for current expanded tree view to display all the tree nodes instead of scrolling down manually.

Video

https://user-images.githubusercontent.com/96312934/184673232-35d23644-2653-44d8-8567-a1bfdd368d59.mov

**Comment by vince-fugnitto** [comment]
@sreeharshakakani did you mean to open this in vscode's repo?

**Comment by vince-fugnitto** [comment]
@sreeharshakakani I think you meant to open this in vscode's [repo](https://github.com/microsoft/vscode)?

**Comment by sreeharshakakani** [comment]
@vince-fugnitto We have Vscode plugins which we are using the same in Theia platform as well, it's the same issue on the UX side. If we have more that one tree view (or) the plugins and expanding the node inside a tree view should either auto scroll down (or) collapse the other views to display the elements on UX properly. Right now, we manually have to scroll down to view all the elements which is not a very good experience to the end user as shown in the video.

**Comment by vince-fugnitto** [comment]
> If we have more that one tree view (or) the plugins and expanding the node inside a tree view should either auto scroll down (or) collapse the other views to display the elements on UX properly.

@sreeharshakakani the behavior is currently consistent with that of vscode (I see you've opened the issue there), performing an auto-scroll or hiding views (how do we determine which view to hide) would also likely cause UX issues (more so than having to scroll manually). 

Given that the behavior is currently consistent, you could always extend the default behavior in your custom Theia-based application if needed.

**Comment by sreeharshakakani** [comment]
@vince-fugnitto yeah, I have opened an issue with vscode as well. I think autoscroll down would be a good option than collapsing. But something which would provide more UX space for viewing the expanded elements would be good.

**Comment by vince-fugnitto** [comment]
Given that the issue was not accepted in VS Code I'm currently marking it as a proposal. If interested you can investigate and contribute a generic implementation to add to the framework. Else, it should always be possible to perform the changes downstream in your application.
